alpine-nix-rails-nginx/etc/nginx/dynamic-content.conf

```diff
@@ -0,0 +1,20 @@
+# Disable caching for dynamic content
+expires off;
+
+# Work-around for BREACH attack: http://breachattack.com/
+gzip off;
+
+# Restrict allowed methods
+if ($request_method !~ ^(DELETE|GET|HEAD|OPTIONS|PATCH|POST|PUT)$) {
+  more_set_headers 'Allow: DELETE,GET,HEAD,OPTIONS,PATCH,POST,PUT';
+  return 405;
+}
+
+# Handle OPTIONS requests
+if ($request_method = OPTIONS) {
```

Comment:
I think that this is a to broad matcher. The available OPTIONS should AFAIK be determined per URL and not generic.

When we see an OPTIONS HTTP verb, only the rails app can know what options are available per endpoint.

Example: `curl -XOPTIONS http://www.example.com/robots.txt` would declare the `robots.txt` to be writable (to anyone, as the request did not include any form of authentication). Maybe we should default to only return `GET/HEAD` here, and ideally only after we know the response is going to an existing URL.